otx/core/data/detection_dataset_adapter.py

```diff
@@ -0,0 +1,86 @@
+"""Detection Dataset Adapter."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+from typing import Any, Tuple
+
+# pylint: disable=invalid-name, too-many-locals, no-member
+from datumaro.components.annotation import Bbox as DatumaroBbox
+from datumaro.components.annotation import Polygon as DatumaroPolygon
+
+from otx.api.entities.annotation import (
+    Annotation,
+    AnnotationSceneEntity,
+    AnnotationSceneKind,
+    NullAnnotationSceneEntity,
+)
+from otx.api.entities.dataset_item import DatasetItemEntity
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.image import Image
+from otx.api.entities.label_schema import LabelSchemaEntity
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.shapes.polygon import Point, Polygon
+from otx.api.entities.shapes.rectangle import Rectangle
+from otx.core.data.base_dataset_adapter import BaseDatasetAdapter
+
+
+class DetectionDatasetAdapter(BaseDatasetAdapter):
+    """Detection adapter inherited by BaseDatasetAdapter.
+
+    It converts DatumaroDataset --> DatasetEntity for object detection, and instance segmentation tasks
+    """
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Detection."""
+        # Prepare label information
+        label_information = self._prepare_label_information(datumaro_dataset)
+        label_entities = label_information["label_entities"]
+
+        # Label schema
+        label_schema = self._generate_default_label_schema(label_entities)
+
+        dataset_items = []
+        for subset, subset_data in datumaro_dataset.items():
+            for _, datumaro_items in subset_data.subsets().items():
+                for datumaro_item in datumaro_items:
+                    image = Image(file_path=datumaro_item.media.path)
+                    shapes = []
+                    for ann in datumaro_item.annotations:
+                        if isinstance(ann, DatumaroPolygon):
```

Comment:
Perfectly, if the datumaro annotation classes have a conversion method like `ann.to_otx_format()` and then we can avoid isinstance here. But since Datumaro knows nothing about otx_sdk, this if else structure can be replaced with a lookup dict: {datumaro annotation type : conversion function}.

Reply:
Make sense, do you think that the current PR should include the lookup dict? or after version?